test/src/main/java/org/springframework/security/test/web/servlet/request/SecurityMockMvcRequestPostProcessors.java

```diff
@@ -1052,6 +1078,176 @@ public MockHttpServletRequest postProcessRequest(MockHttpServletRequest request)
+		 *
+		 * @param authorities the authorities to use
+		 * @return the {@link OidcLoginRequestPostProcessor} for further configuration
+		 */
+		public OidcLoginRequestPostProcessor authorities(Collection<GrantedAuthority> authorities) {
+			Assert.notNull(authorities, "authorities cannot be null");
+			this.authorities = authorities;
+			return this;
+		}
+
+		/**
+		 * Use the provided authorities in the {@link Authentication}
+		 *
+		 * @param authorities the authorities to use
+		 * @return the {@link OidcLoginRequestPostProcessor} for further configuration
+		 */
+		public OidcLoginRequestPostProcessor authorities(GrantedAuthority... authorities) {
+			Assert.notNull(authorities, "authorities cannot be null");
+			this.authorities = Arrays.asList(authorities);
+			return this;
+		}
+
+		/**
+		 * Use the provided {@link OidcIdToken} when constructing the authenticated user
+		 *
+		 * @param idTokenBuilderConsumer a {@link Consumer} of a {@link OidcIdToken.Builder}
+		 * @return the {@link OidcLoginRequestPostProcessor} for further configuration
+		 */
+		public OidcLoginRequestPostProcessor idToken(Consumer<OidcIdToken.Builder> idTokenBuilderConsumer) {
+			OidcIdToken.Builder builder = OidcIdToken.withTokenValue("id-token");
+			builder.subject("test-subject");
+			idTokenBuilderConsumer.accept(builder);
+			this.idToken = builder.build();
+			return this;
+		}
+
+		/**
+		 * Use the provided {@link OidcUserInfo} when constructing the authenticated user
+		 *
+		 * @param userInfoBuilderConsumer a {@link Consumer} of a {@link OidcUserInfo.Builder}
+		 * @return the {@link OidcLoginRequestPostProcessor} for further configuration
+		 */
+		public OidcLoginRequestPostProcessor userInfoToken(Consumer<OidcUserInfo.Builder> userInfoBuilderConsumer) {
+			OidcUserInfo.Builder builder = OidcUserInfo.builder();
+			userInfoBuilderConsumer.accept(builder);
+			this.userInfo = builder.build();
+			return this;
+		}
+
+		/**
+		 * Use the provided {@link OidcUser} as the authenticated user.
+		 *
+		 * Supplying an {@link OidcUser} will take precedence over {@link #idToken}, {@link #userInfo},
+		 * and list of {@link GrantedAuthority}s to use.
+		 *
+		 * @param oidcUser the {@link OidcUser} to use
+		 * @return the {@link OidcLoginRequestPostProcessor} for further configuration
+		 */
+		public OidcLoginRequestPostProcessor oidcUser(OidcUser oidcUser) {
+			this.oidcUser = oidcUser;
+			return this;
+		}
+
+		@Override
+		public MockHttpServletRequest postProcessRequest(MockHttpServletRequest request) {
+			OidcUser oidcUser = getOidcUser();
+			OAuth2AuthenticationToken token = new OAuth2AuthenticationToken(oidcUser, oidcUser.getAuthorities(), this.clientRegistration.getRegistrationId());
+			OAuth2AuthorizedClient client = new OAuth2AuthorizedClient(this.clientRegistration, token.getName(), this.accessToken);
+
+			ClientRegistrationRepository clients = new InMemoryClientRegistrationRepository(this.clientRegistration);
```

Comment:
Again regarding multi-threading, @rwinch, I believe this to be thread-safe since it is not mutable state, e.g. there's no save method on `ClientRegistrationRepository` and the tester can't supply a `ClientRegistration` instance.

However, if these features were added, I'd be at a loss for how to properly inject a `ClientRegistrationRepository` in a thread-safe manner.